api/schemas/timelog.py

```diff
@@ -12,23 +12,46 @@ class Config:
         orm_mode = True
 
 
-class Template(BaseModel):
-    id: int
-    name: str
-    story: Optional[str]
-    description: Optional[str]
-    task_type: Optional[str]
+# Shared properties
+class TemplateBase(BaseModel):
+    name: Optional[constr(max_length=80)]
+    story: Optional[constr(max_length=80)]
+    description: Optional[constr(max_length=8192)]
+    task_type: Optional[constr(max_length=40)]
     init_time: Optional[int]
     end_time: Optional[int]
     customer_id: Optional[int]
     user_id: Optional[int]
     project_id: Optional[int]
-    is_global: bool
+    is_global: Optional[bool]
+
+    @root_validator(pre=True)
+    def user_template_not_global(cls, values):
+        user_id, is_global = values.get("user_id"), values.get("is_global")
+        if is_global is not None and user_id is not None and is_global is True:
+            raise ValueError("Global templates should not have a user_id.")
```

Comment:
The check can be simplified a bit and we also need to cover when is not global and doesn't have user, to avoid having "abandoned" templates:

```suggestion
        if is_global and user_id is not None:
            raise ValueError("Global templates should not have a user_id.")
        elif not is_global and not user_id:
            raise ValueError("Private templates should have a user_id.")
```

Reply:
Great point. Fixing.